nova/tests/unit/scheduler/filters/test_bigvm_filters.py

```diff
@@ -231,141 +232,110 @@ def call(a, b):
         self.assertTrue(call(1024, 512))
         self.assertFalse(call(1024, 511))
 
-    def test_big_vm_with_matching_full_size(self):
-        """Test automatic full size matching."""
+    def test_big_vm_with_missing_extra_specs(self):
+        """If the extra_specs flavor attr is missing, we fail."""
         spec_obj = objects.RequestSpec(
-            flavor=objects.Flavor(memory_mb=CONF.bigvm_mb, extra_specs={}))
+            flavor=objects.Flavor(memory_mb=CONF.bigvm_mb))
         host = fakes.FakeHostState('host1', 'compute',
-                {'uuid': uuidsentinel.host1})
-        self.assertTrue(self.filt_cls.host_passes(host, spec_obj))
+                                   {'uuid': uuidsentinel.host1})
+        self.assertRaises(ObjectActionError, self.filt_cls.host_passes,
+                          host, spec_obj)
```

Comment:
Honestly, I don't know if this might _reasonably_ happen or not. It seemed like a valid thing to check for. I will see if I can find out whether this seems at all plausible by looking for calling code...

Reply:
But if an exception means the whole scheduler dies, then this needs to be caught (if it's conceivable) or this test removed (if it's not likely to occur).

Reply:
This looks helpful:
https://github.com/sapcc/nova/blob/ed2db87278b102a8be34151bccaab81458b96b1c/nova/scheduler/filters/aggregate_instance_extra_specs.py#L45-L49

Reply:
Yes doing it this way would consolidate it with that class. I will do that in a separate commit, sometime later... For now I'll just remove the check in `host_passes()` -> smaller diff.

Reply:
Removed the whole test.